neo/SmartContract/InteropService.cs

```diff
@@ -499,6 +511,15 @@ private static bool Contract_Call(ApplicationEngine engine)
             if (currentManifest != null && !currentManifest.CanCall(contract.Manifest, method.GetString()))
                 return false;
 
+            if (engine.InvocationCounter.TryGetValue(contract.ScriptHash, out var counter))
```

Comment:
https://github.com/neo-project/neo/blob/e497d95c1aa446c319cd7dbae1b674c882c052cd/neo/SmartContract/ApplicationEngine.cs#L54-L57

I think we can move the counter logic into `ApplicationEngine_ContextLoaded()`.